soccer/gameplay/fsm.py

```diff
@@ -81,25 +96,34 @@ def spin(self):
             StateMachine.spin(self)
 
     # if you add a transition that already exists, the old one will be overwritten
-    def add_transition(self, from_state, to_state,
-                       condition: Union[bool, Callable],
-                       event_name: str):
+    def add_transition(self, from_state: State, to_state: State,
+                       condition: Union[bool, TransitionFunction],
+                       event_name: str) -> None:
         if isinstance(condition, bool):
-            condition = lambda: condition
+            condition_bool: bool = cast(bool, condition)
+
+            def condition_fn() -> bool:
+                return condition_bool
```

Comment:
For clarity could you make this a lambda? I feel like declaration is less clear than assignment. Also, you already know condition is a bool from the isinstance check so there should be no need for the cast.

Reply:
`mypy` disagrees : (

Reply:
What the hell? Why does mypy say no lambdas here? We aren't getting rid of lambdas everywhere in our code just to make mypy happy...

Reply:
If it's just typing concerns this answer: https://stackoverflow.com/questions/33833881/is-it-possible-to-type-hint-a-lambda-function seems to give a direction.

You would just have to let `condition: Callable[[], bool] = lambda: condition_bool` and then you should be good, right? Or am I missing something?

Reply:
ok wtf it works now